mesonbuild/dependencies/mpi.py

```diff
@@ -146,53 +172,39 @@ def _is_link_arg(self, f: str) -> bool:
                     f == '-pthread' or
                     (f.startswith('-W') and f != '-Wall' and not f.startswith('-Werror')))
 
-
-class IntelMPIConfigToolDependency(_MPIConfigToolDependency):
-
-    """Wrapper around Intel's mpiicc and friends."""
-
-    version_arg = '-v'  # --version is not the same as -v
-
-    def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any],
-                 language: T.Optional[str] = None):
-        super().__init__(name, env, kwargs, language=language)
-        if not self.is_found:
-            return
-
-        args = self.get_config_value(['-show'], 'link and compile args')
-        self.compile_args = self._filter_compile_args(args)
-        self.link_args = self._filter_link_args(args)
-
-    def _sanitize_version(self, out: str) -> str:
-        v = re.search(r'(\d{4}) Update (\d)', out)
+    def _check_and_get_version(self, tool: T.List[str], returncode: int) -> T.Tuple[bool, T.Union[str, None]]:
+        p, out = Popen_safe(tool + ["--showme:version"])[:2]
+        valid = p.returncode == returncode
+        if valid:
+            # OpenMPI
+            v = re.search(r'\d+.\d+.\d+', out)
+            if v:
+                version = v.group(0)
+            else:
+                version = None
+            return valid, version
+
+        # --version is not the same as -v
+        p, out = Popen_safe(tool + ["-v"])[:2]
+        valid = p.returncode == returncode
+        first_line = out.split("\n", maxsplit=1)[0]
```

Comment:
These double quotes are inconsistent and should be single quotes.

Reply:
I changed the double quotes as single quotes.